pkg/auth/tokenparser.go

```diff
@@ -21,20 +22,18 @@ type TokenClaims struct {
 
 // TokenParser represents a parser for JWT tokens.
 type TokenParser struct {
-	logger     *log.Logger
 	keyManager *KeyManager
 }
 
 // NewTokenParser creates a new TokenParser.
-func NewTokenParser(logger *log.Logger, keyManager *KeyManager) (*TokenParser, error) {
-	if logger == nil {
+func NewTokenParser(keyManager *KeyManager) (*TokenParser, error) {
+	if log.Logger() == nil {
```

Comment:
I don't understand why we need this check, because:
* it should have been set in the main, so we can easily panic - if it is not set then it means that there is something really really wrong that shouldn't happen
* if we don't want to fail, then we can lazily create the instance of the logger if it is nil - in the same way as it is done in the main via `log.InitializeLogger("logger")`

Reply:
fixed here: f056abe3a4ff3aa5909f9272b03bb4bb63d732b0